documentation/source/Segmentation.md

```diff
@@ -0,0 +1,152 @@
+
+We have images and labels, with the default batch size of 256 for training.
+```py
+from PIL import Image
+from torchvision.utils import draw_segmentation_masks
+from torchvision.transforms import ToTensor, ToPILImage, Resize
+import numpy as np
+import torch
+
+def plot_seg_data(img_path: str, target_path: str):
+  image = (ToTensor()(Image.open(img_path).convert('RGB')) * 255).type(torch.uint8)
+  target = torch.from_numpy(np.array(Image.open(target_path))).bool()
+  image = draw_segmentation_masks(image, target, colors="red", alpha=0.4)
+  image = Resize(size=200)(image)
+  display(ToPILImage()(image))
+
+for i in range(4, 7):
+  img_path, target_path = train_loader.dataset.samples_targets_tuples_list[i]
+  plot_seg_data(img_path, target_path)
+```
+![segmentation_target.png](segmentation_target.png)
+
+
+## Load the model from modelzoo
+
+Create a PPLiteSeg nn.Module, with 1 class segmentation head classifier. For simplicity `use_aux_head` is set as `False`
+and extra Auxiliary heads aren't used for training.
+
+```py
+from super_gradients.training import models
+from super_gradients.common.object_names import Models
+
+model = models.get(
+    model_name=Models.PP_LITE_T_SEG75,      # You can use any model listed in the Models.<Name>
+    arch_params={"use_aux_heads": False},
+    num_classes=1,                          # Change this if you work on another dataset with more classes
+    pretrained_weights="cityscapes"         # Drop this line to train from scratch
+)
+```
+Notes:
+- SG includes implementations of 
+[many different architectures](https://github.com/Deci-AI/super-gradients#implemented-model-architectures).
+- Most of these architectures have [pretrained checkpoints](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/Computer_Vision_Models_Pretrained_Checkpoints.md) so feel free to experiment!
+
+
+
+### Setup training parameters
+The training parameters includes loss, metrics, learning rates and much more. You can check out the [default training parameters](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/recipes/training_hyperparams/default_train_params.yaml).
+For this task, we will train for 30 epoch, using Binary IoU using the SGD optimizer.
+```py
+from super_gradients.training.metrics.segmentation_metrics import BinaryIOU
+
+train_params = {
+    "max_epochs": 30,
+    "lr_mode": "cosine",
+    "initial_lr": 0.005,
+    "lr_warmup_epochs": 5,
+    "multiply_head_lr": 10,
+    "optimizer": "SGD",
+    "loss": "bce_dice_loss",
+    "ema": True,
+    "zero_weight_decay_on_bias_and_bn": True,
+    "average_best_models": True,
+    "metric_to_watch": "target_IOU",
+    "greater_metric_to_watch_is_better": True,
+    "train_metrics_list": [BinaryIOU()],
+    "valid_metrics_list": [BinaryIOU()],
+    "loss_logging_items_names": ["loss"],
+}
+```
+
+
+### Launch Training
+Now you only need to setup a Trainer and launch the training!
```

Comment:
This might be a good place to shortly discuss the Trainer concept, which is not native to torch.
"The Trainer is a class that...."  
"In the example below, we will call the `train()` function, which is the main entry point for any training process... "

Reply:
Changed in d1fb7168b1580600f5594a42819fa8246042c207